src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -841,7 +840,12 @@ private void ConfigureLogging()
             // to the startup logger until we've read configuration settings and can create the real logger.
             // The "startup" logger is used in this class for startup related logs. The public logger is used
             // for all other logging after startup.
-            FunctionTraceWriterFactory = new FunctionTraceWriterFactory(ScriptConfig);
+            FunctionTraceWriterFactory = new FunctionTraceWriterFactory(ScriptConfig, t =>
+            {
+                // File logging is done conditionally.
+                return FileLoggingEnabled && (!(t.Properties?.ContainsKey(ScriptConstants.TracePropertyPrimaryHostKey) ?? false) || IsPrimary);
+            });
```

Comment:
this logic is a little hard to follow, between the negation, the or, and the null coalesce. Can you introduce one or more two named variables?